### PR TITLE
Do not abort bufr reading if message has invalid date information

### PR DESCRIPTION
**Pull request by trygveasp**

### CODE REVIEW

File: pysurfex/bufr.py [hunk 1]
```diff
@@ -66,6 +65,9 @@ def __init__(
 
         # open bufr file
         file_handler = open(bufrfile, mode="rb")
+        lines_in_file = file_handler.readlines()
+        number_of_lines = int(len(lines_in_file))
```
Comment (joewkr) [inline pysurfex/bufr.py:69]: Looks a bit strange to see `readlines` for a file opened in the binary mode. Why not using a combination of `seek` and `tell` to get the file size and then `tell` in the loop to derive the percentage of processed data?

**Review by joewkr — CHANGES_REQUESTED** [review]

File: pysurfex/bufr.py [hunk 2]
```diff
@@ -384,18 +405,24 @@ def __init__(
...
+                    nbytes = number_of_bytes
 
-                if (cnt % 1000) == 0:
-                    print(".", end="")
-                    sys.stdout.flush()
+                processed = int(round(float(nbytes) * 100.0 / float(number_of_bytes)))
+                if processed not in processed_threshold:
+                    if processed % 5 == 0:
+                        processed_threshold.append(processed)
+                        logging.info("Read: %s%%", processed)
```
Comment (joewkr) [inline pysurfex/bufr.py:417]: Do you really need a list there? Wouldn't it suffice to have something like:
```python
if processed > processed_threshold and processed % 5 == 0:
    processed_threshold = processed
    logging.info("Read: %s%%", processed)
```

File: pysurfex/bufr.py [hunk 3]
```diff
@@ -83,6 +85,7 @@ def __init__(
             "stationNumber",
             "blockNumber",
         ]
+        processed_threshold = [0]
```
Comment (joewkr) [inline pysurfex/bufr.py:88]: See below

**Review by joewkr — CHANGES_REQUESTED** [review]

**Comment by trygveasp** [comment]
Should be addressed now?

**Review by joewkr — APPROVED** [review]
LGTM

**Pull request merged**